File: app/src/main/java/com/example/shibeapplication/adapters/ShibeAdapter.java
package com.example.shibeapplication.adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.shibeapplication.R;
import com.squareup.picasso.Picasso;

import java.util.List;

public class ShibeAdapter extends RecyclerView.Adapter<ShibeAdapter.ShibeViewHolder> {
    private Context context;
    private List<String> shibeURLs;

    public ShibeAdapter(List<String> urls) {
        shibeURLs = urls;
    }

    @NonNull
    @Override
    public ShibeViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        context = parent.getContext();

        View theView = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.shibe_item, parent, false);
        return new ShibeViewHolder(theView);
    }

    @Override
    public void onBindViewHolder(@NonNull ShibeViewHolder holder, int position) {
        //TOdo bind the image views to the data received from the list.

        String imageString = shibeURLs.get(position);

        Picasso.get()
                .load(imageString)
                .fit()
                .into(holder.shibeImage);


    }

    @Override
    public int getItemCount() {
        return shibeURLs.size();
    }

    public class ShibeViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {

        ImageView shibeImage;

        public ShibeViewHolder(@NonNull View itemView) {
            super(itemView);
            shibeImage = itemView.findViewById(R.id.rv_shibeImage);
            shibeImage.setOnClickListener(this);

        }

        @Override
        public void onClick(View view) {
            //TODO This is where you should make the position of the image show up
        }
    }
}

File: app/src/main/java/com/example/shibeapplication/views/adapters/ShibeAdapter.kt
package com.example.shibeapplication.views.adapters

import android.content.Context
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.Toast
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.RecyclerView.ViewHolder
import com.example.shibeapplication.R
import com.example.shibeapplication.databinding.ShibeItemBinding
import com.example.shibeapplication.views.adapters.ShibeAdapter.ShibeViewHolder
import com.squareup.picasso.Picasso
import java.util.zip.Inflater

class ShibeAdapter(private val shibeURLs: List<String>) : RecyclerView.Adapter<ShibeViewHolder>() {
    private var context: Context? = null
    private lateinit var binding: ShibeItemBinding

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ShibeViewHolder {
        binding = ShibeItemBinding.inflate(
                LayoutInflater.from(parent.context), parent, false)

        return ShibeViewHolder(binding)
    }

    override fun onBindViewHolder(holder: ShibeViewHolder, position: Int) {
        val imageString = shibeURLs[position]
        val imagePosition = position.toString()

        holder.bindItem(imageString, imagePosition)
    }

    override fun getItemCount(): Int {
        return shibeURLs.size
    }

    inner class ShibeViewHolder(binding: ShibeItemBinding) : ViewHolder(binding.root), View.OnClickListener {
        override fun onClick(view: View) {
            Toast.makeText(context, view.tag.toString(), Toast.LENGTH_SHORT).show()
            Log.e("POSITION OF IMAGE", view.tag.toString())

            binding.rvShibeImage.setOnClickListener(this)
        }

        fun bindItem(image: String, position: String) {
            binding.rvShibeImage.tag = "Image position in array was :$position"
            Picasso.get()
                    .load(image)
                    .fit()
                    .into(binding.rvShibeImage)
        }
    }
}